torchvision/datasets/triplet.py

```diff
@@ -0,0 +1,139 @@
+from .folder import DatasetFolder
+
+from torch.distributions.multinomial import Multinomial
+from torch.distributions.bernoulli import Bernoulli
+
+import torch
+import torch.utils.data as data
+
+import math
+
+
+def get_class_samples(num_classes, samples):
+    """Bins samples with respect to classes
+
+    Args:
+        num_classes (int): number of classes
+        samples (list[tuple]): (sample, class_idx) tuples
+
+    Returns:
+        list[list]: Bins of sample paths, binned by class_idx
+    """
+    class_samples = [[] for _ in range(num_classes)]
+    for sample_path, class_idx in samples:
+        class_samples[class_idx].append(sample_path)
+
+    return class_samples
+
+
+def get_sampled_indices(dist, num_samples):
+    """
+    Samples from distribution till the number of the non-zero entries in the sample
+    equals num_samples
+
+    Args:
+        dist (torch.distributions.distribution.Distribution): A pytorch distribution
+        num_samples (int): number of non-zero samples needed
+
+    Returns:
+        list: indices of the non-zero samples
+    """
+    sample = dist.sample().nonzero().view(-1)
+    while len(sample) != num_samples:
+        sample = dist.sample().nonzero().view(-1)
+
+    return sample.tolist()
+
+
+def generate_triplets(class_samples, num_triplets):
+    """Generates a set of triplets from bins of samples
+
+    Args:
+        class_samples (list[list]): bins of samples, binned by class
+        num_triplets (int): number of triplets to be generated
+
+    Returns:
+        list[tuple]: triplets of the form (anchor, positive, negative)
+    """
+    triplets = []
+    class_dist = Multinomial(2, torch.Tensor([1.] * len(class_samples)))
```

Comment:
Do we need to use `torch.distributions` here? Can't we just use `torch.multinomial` and maybe even the `random` package instead?

Reply:
That's exactly what I was looking for! You're right, we don't need torch.distributions. I'll change this right away.

Reply:
I have removed `torch.distributions` and used `torch.multinomial` wherever it was required.